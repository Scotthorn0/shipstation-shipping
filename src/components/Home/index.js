import React, { Component } from 'react';
import { Menu, Container, Header } from 'semantic-ui-react';
import Product from '../Product';
import Ship from '../Ship';
import './styles.css';

class Home extends Component {
  state = {
    activeItem: 'product'
  }

  handleMenuClick = (_, { name: activeItem }) => {
    this.setState({
      activeItem,
    })
  }

  renderChild = (activeItem) => (
    activeItem === 'product' ?
      <Product /> :
      <Ship />
  )

  render() {
    const { activeItem } = this.state;

    return (
      <Container>
        <Header
          content="Welcome"
        />
        <Menu tabular>
          <Menu.Item name="product" active={activeItem === "product"} onClick={this.handleMenuClick}/>
          <Menu.Item name="ship" active={activeItem === "ship"} onClick={this.handleMenuClick}/>
        </Menu>
        {this.renderChild(activeItem)}
      </Container>
    );
  }
}

export default Home;
