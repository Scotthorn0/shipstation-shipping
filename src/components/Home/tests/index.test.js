import React from 'react';
import { shallow } from 'enzyme';
import Home from '../';

describe('Home Component', () => {
  let wrapper;
  let instance;

  beforeEach(() => {
    wrapper = shallow(<Home />);
    instance = wrapper.instance();
  });

  test('should render without crashing', () => {
    expect(wrapper.exists()).toBe.true;
  });

  test('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should default to rendering the Product page', () => {
    expect(wrapper.state('activeItem')).toBe('product');
    expect(wrapper.find('Product').length).toBe(1);
    expect(wrapper.find('Ship').length).toBe(0);
  });

  it('should have a MenuClick handler that changes state', () => {
    // Find the correct Menu Item and simulate a click with a mock event object
    wrapper.find('[name="ship"]').simulate('click', null, { name: 'ship'});

    expect(wrapper.state('activeItem')).toBe('ship');
    expect(wrapper.find('Product').length).toBe(0);
    expect(wrapper.find('Ship').length).toBe(1);
  });
});
