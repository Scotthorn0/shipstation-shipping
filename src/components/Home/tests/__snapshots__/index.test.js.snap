// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Header
          content="Welcome"
/>,
        <Menu
          tabular={true}
>
          <MenuItem
                    active={true}
                    name="product"
                    onClick={[Function]}
          />
          <MenuItem
                    active={false}
                    name="ship"
                    onClick={[Function]}
          />
</Menu>,
        <Product />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "content": "Welcome",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <MenuItem
              active={true}
              name="product"
              onClick={[Function]}
/>,
            <MenuItem
              active={false}
              name="ship"
              onClick={[Function]}
/>,
          ],
          "tabular": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "active": true,
              "name": "product",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "active": false,
              "name": "ship",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Header
            content="Welcome"
/>,
          <Menu
            tabular={true}
>
            <MenuItem
                        active={true}
                        name="product"
                        onClick={[Function]}
            />
            <MenuItem
                        active={false}
                        name="ship"
                        onClick={[Function]}
            />
</Menu>,
          <Product />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "content": "Welcome",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <MenuItem
                active={true}
                name="product"
                onClick={[Function]}
/>,
              <MenuItem
                active={false}
                name="ship"
                onClick={[Function]}
/>,
            ],
            "tabular": true,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "active": true,
                "name": "product",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "active": false,
                "name": "ship",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
